src/qubic.cpp

```diff
@@ -1110,6 +1110,7 @@ static void requestProcessor(void* ProcedureArgument)
     RequestResponseHeader* header = (RequestResponseHeader*)processor->buffer;
     while (!shutDownNode)
     {
+        score->tryProcessSolution(processorNumber);

```

Comment:
I would add a comment here like "Process a solutions if any is queued. This is for parallelizing and is triggered by tick processing." (because this is not obvious)

Reply:
forgot to change this part, this should be:
```
if (processorNumber < NUMBER_OF_SOLUTION_PROCESSORS)
{
    score->tryProcessSolution(processorNumber);
}
```

which will also solve the lock you mentioned here: https://github.com/qubic/core/pull/39#discussion_r1461634159

or `(processorNumber - 2) < NUMBER_OF_SOLUTION_PROCESSORS`? I'm not sure where exactly the index starts for requestProcessor

Reply:
This would be a quite easy solution, but I think with this we may end up using less than NUMBER_OF_SOLUTION_PROCESSORS for processing solutions if the enabled or health bit are not set for some of the processors.